lib/Compilation.js

```diff
@@ -955,11 +955,14 @@ class Compilation {
 			},
 			err => {
 				if (this.bail) {
+					const result = callback(err);
+
 					this.buildQueue.stop();
 					this.rebuildQueue.stop();
 					this.processDependenciesQueue.stop();
 					this.factorizeQueue.stop();
-					return callback(err);
+
+					return result;
```

Comment:
```suggestion
					return;
```

`return callback(err);` is only used as shortcut for `callback(err); return;`. We don't care about the result value.

Reply:
fixed